---
uint32 chip_id